test/lib/fpti-test-utils.js

```diff
@@ -0,0 +1,74 @@
+/* @flow */
+const { expect } = global;
+
+export const config = {
+  containerSummary: {
+    programId: 'A87KPP4KFVC8J',
+    mrid: 'JTJ8GTMKP7V3A'
+  }
+};
+
+export const deviceInfo = {
+  deviceWidth: 123,
+  deviceHeight: 125,
+  screenWidth: 300,
+  screenHeight: 400,
+  colorDepth: 24,
+  rosettaLanguage: 'en-US',
+  location: 'test.business.us',
+  deviceType: 'Desktop',
+  browserHeight: 100,
+  browserWidth: 300
+};
+
+export const identity = {
+  confidenceScore: 100,
+  identificationType: 'RMUC',
+  encryptedAccountNumber: 'CY26RACMGXKQL'
+};
+
+export const generatedUserIds = {
+  userId: 'ee964537-1c7b-403e-b978-ea29babc5aed',
+  merchantProvidedUserId: '4328f923-3293-4b24-a069-d3c0bc2a0375'
+};
+
+export function verifyMainFptiAttributes(fptiEvent, input) {
+  const { eventName, payload } = input;
+  const eventDataPayload = { ...payload };
+  const userId = eventDataPayload.user_id;
+  delete eventDataPayload.user_id;
+
+  expect(fptiEvent.eventName).toEqual(eventName);
+  expect(fptiEvent.eventType).toEqual(eventName);
+  expect(fptiEvent.eventData).toEqual(JSON.stringify(eventDataPayload));
+  expect(fptiEvent.e).toEqual('im');
+  expect(fptiEvent.page).toEqual(`ppshopping:${ eventName }`);
+  expect(fptiEvent.merchantProvidedUserId).toEqual(userId);
+}
+
+export function verifyFptiIdentityMappings(fptiEvent, identityData) {
+  expect(fptiEvent.confidenceScore).toEqual(identityData.confidenceScore);
+  expect(fptiEvent.encryptedAccountNumber).toEqual(
+    identityData.encryptedAccountNumber
+  );
+  expect(fptiEvent.identificationType).toEqual(identityData.identificationType);
+}
+
+export function verifyFptiDeviceInfoMappings(fptiEvent, deviceInfoInput) {
```

Comment:
Why didn't you use following assert?
```
expect(fptiEvent).toEqual(deviceInfoInput);
```

Reply:
only the subset of fptiEvent attributes are mapped from deviceInfoInput